### PR TITLE
[ENHANCEMENT] Allow findByAria to match regex first

### PR DESCRIPTION
**Pull request by wei-tg**
if is an instance of String, use match Regex

### CODE REVIEW

File: src/finders/find-by-aria/index.js [hunk 1]
```diff
@@ -1,6 +1,19 @@
...
+function filterCondition(element, labelText) {
+  if (computeAria(element) instanceof String || typeof computeAria(element) === 'string') {
+    return matchRegex(computeAria(element));
+  }
+  return computeAria(element) === labelText.toLowerCase();
+}
+
 export default function findByAria(selector, labelText) {
-  return findAll(selector).filter(element => computeAria(element) === labelText.toLowerCase());
+  return findAll(selector).filter(element => filterCondition(element, labelText));
```
Comment (billybonks) [inline src/finders/find-by-aria/index.js]: what we actually want to do is

if labelText is a regex then return`labelText.test(computeAria(element))`

Comment (wei-tg) [reply src/finders/find-by-aria/index.js]: sorry i got confused with the requirements 🐤

**Comment by wei-tg** [comment]
use helper: matchLabel in all existing finders

File: src/finders/find-by-name.js [hunk 2]
```diff
@@ -1,7 +1,19 @@
 import findAll from './helpers/findAll';
+import matchLabel from './helpers/matchLabel';
 
 function findByName(selector, text) {
   const elements = selector.split(',');
+
+  if (text instanceof RegExp) {
+    const foundInputs = findAll(`${elements.join('[name],')}[name]`);
+    const results = [];
+    for (var i in foundInputs) {
```
Comment (billybonks) [inline src/finders/find-by-name.js]: can use `foundInputs.filter` here instead of for loop and results array

Comment (wei-tg) [reply src/finders/find-by-name.js]: donez!

File: __test__/finders/find-by-name.test.js [hunk 3]
```diff
@@ -8,7 +8,9 @@ describe('Find by name', () => {
       <div id="hen" name="hello"><div>
       <span id="bu" name="hello"><span>
       <input id="hao" name="world" />
+      <input id="regexz" name="herloo /^[$A-Za-z_][0-9A-Za-z_$-:]*$/<div>i did bad thing </div>" />
     `;
+    //
```
Comment (billybonks) [inline __test__/finders/find-by-name.test.js]: 🔥

Comment (wei-tg) [reply __test__/finders/find-by-name.test.js]: 😅

File: src/finders/find-by-name.js [hunk 4]
```diff
@@ -1,7 +1,15 @@
 import findAll from './helpers/findAll';
+import matchLabel from './helpers/matchLabel';
 
 function findByName(selector, text) {
   const elements = selector.split(',');
+
+  if (text instanceof RegExp) {
+    const foundInputs = findAll(`${elements.join('[name],')}[name]`);
```
Comment (billybonks) [inline src/finders/find-by-name.js]: lets add a comment here explaining why we are only finding elements with a name attribute

**Comment by billybonks** [comment]
Lints are failing

**Pull request merged**